aws/s3.go

```diff
@@ -0,0 +1,200 @@
+package aws
+
+import (
+	"context"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/skit-ai/vcore/errors"
+	"github.com/skit-ai/vcore/log/slog"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+const (
+	// Regex for S3 URLs, VPCE interface endpoint
```

Comment:
Can you add reference for this?

Reply:
[ref](https://github.com/aws/amazon-ssm-agent/blob/master/agent/s3util/s3uri.go)

Reply:
Can add this as a comment here

Reply:
added